Build publishing github action is failing

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The [Push Image Build](https://github.com/rook/rook/actions/workflows/push-build.yaml) github workflow is failing for every master and release branch for the past 2-3 days. 

This is blocking both the docs from being published, and the docker images from being pushed to dockerhub. 

The errors show that pip is failing to install the `mkdocs-exclude` package when preparing to publish the docs.

```
pip3 install -r requirements_docs.txt
Defaulting to user installation because normal site-packages is not writeable
Collecting mike (from -r requirements_docs.txt (line 1))
  Downloading mike-2.1.3-py3-none-any.whl.metadata (17 kB)
Collecting mkdocs (from -r requirements_docs.txt (line 2))
  Downloading mkdocs-1.6.1-py3-none-any.whl.metadata (6.0 kB)
Collecting mkdocs-awesome-pages-plugin (from -r requirements_docs.txt (line 3))
  Downloading mkdocs_awesome_pages_plugin-2.10.1-py3-none-any.whl.metadata (13 kB)
Collecting mkdocs-exclude (from -r requirements_docs.txt (line 4))
  Downloading mkdocs-exclude-1.0.2.tar.gz (6.8 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [43 lines of output]
      running egg_info
      creating /tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info
      writing /tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info/dependency_links.txt
      writing entry points to /tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info/entry_points.txt
      writing requirements to /tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info/requires.txt
      writing top-level names to /tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info/top_level.txt
      writing manifest file '/tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info/SOURCES.txt'
      reading manifest file '/tmp/pip-pip-egg-info-ehulma8o/mkdocs_exclude.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-eubewme4/mkdocs-exclude_758af234ddcf457a9a1a48dac0fa1ec3/setup.py", line 10, in <module>
          setuptools.setup(
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/__init__.py", line 117, in setup
          return distutils.core.setup(**attrs)
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 186, in setup
          return run_commands(dist)
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
          dist.run_commands()
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 983, in run_commands
          self.run_command(cmd)
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 312, in run
          self.find_sources()
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 320, in find_sources
          mm.run()
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 548, in run
          self.prune_file_list()
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/command/sdist.py", line 162, in prune_file_list
          super().prune_file_list()
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/_distutils/command/sdist.py", line 380, in prune_file_list
          base_dir = self.distribution.get_fullname()
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/_core_metadata.py", line 272, in get_fullname
          return _distribution_fullname(self.get_name(), self.get_version())
        File "/home/runner/.local/lib/python3.10/site-packages/setuptools/_core_metadata.py", line 290, in _distribution_fullname
          canonicalize_version(version, strip_trailing_zero=False),
      TypeError: canonicalize_version() got an unexpected keyword argument 'strip_trailing_zero'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
make: *** [Makefile:129: deps.docs] Error 1
```

I am not able to repro this on my mac in a python virtual env. The same command succeeds, here is a snippet:
```
% pip3 install -r requirements_docs.txt
Collecting mike (from -r ../requirements_docs.txt (line 1))
  Downloading mike-2.1.3-py3-none-any.whl.metadata (17 kB)
Collecting mkdocs (from -r ../requirements_docs.txt (line 2))
  Downloading mkdocs-1.6.1-py3-none-any.whl.metadata (6.0 kB)
Collecting mkdocs-awesome-pages-plugin (from -r ../requirements_docs.txt (line 3))
  Downloading mkdocs_awesome_pages_plugin-2.10.1-py3-none-any.whl.metadata (13 kB)
Collecting mkdocs-exclude (from -r ../requirements_docs.txt (line 4))
  Downloading mkdocs-exclude-1.0.2.tar.gz (6.8 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
...
```

**Expected behavior:**
Published builds must succeed.

--- a/.github/workflows/push-build.yaml
+++ b/.github/workflows/push-build.yaml
@@ -54,6 +54,10 @@ jobs:
           echo "BRANCH_NAME=${GITHUB_REF##*/}" >> $GITHUB_ENV
           echo "GITHUB_REF"=${GITHUB_REF} >> $GITHUB_ENV
 
+      - uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+        with:
+          python-version: 3.13
+
       - name: Install dependencies
         run: |
           python3 -m pip install --upgrade pip setuptools wheel

